cilium-health/launch/endpoint.go

```diff
@@ -305,7 +306,7 @@ func LaunchAsEndpoint(owner endpoint.Owner, n *node.Node, mtuConfig mtu.Configur
 		Reason: "health daemon bootstrap",
 	})
 	if !buildSuccessful {
-		endpointmanager.Remove(ep)
+		endpointmanager.Remove(ep, &sync.WaitGroup{})
```

Comment:
This should be nil - alternatively create two separate 'Remove' functions (one that asynchronously releases endpoint ID, one that doesn't).